File: spec/lua/extra/custom_fx.lua
local class = require("class")

Nested = {}
Nested.Deeply = {}
Nested.Deeply.CustomFx = class.class()

function Nested.Deeply.CustomFx:_init(foo)
end

function Nested.Deeply.CustomFx:decode(data)
    return "_" .. data .. "_"
end

File: spec/php/extra/CustomFx.php
<?php
namespace Kaitai\Struct\Tests;

class MyCustomFx {
    public function __construct($key, $flag, $someBytes) {
        $this->key = $flag ? $key : -$key;
    }

    public function decode($src) {
        $dst = '';
        for ($i = 0, $n = strlen($src); $i < $n; $i++) {
            $dst .= chr(ord($src[$i]) + $this->key);
        }
        return $dst;
    }
}

class CustomFxNoArgs {
    public function __construct() {
    }

    public function decode($src) {
        return "_" . $src . "_";
    }
}

namespace Nested\Deeply;

class CustomFx {
    public function __construct(int $x) {
    }

    public function decode($src) {
        return "_" . $src . "_";
    }
}
